### ISSUE TITLE
pointcloud data in pnts good, but gltf with EXT_structural_metadata  failed to render the same

### ISSUE THREAD

**Issue by geo3data** [CesiumGS/cesium#11345]
hi,
we have C++ tools that process point cloud to 3dtiles pnts format,  it works fine.  Do some special effects with the attributes of the point cloud， eg,  display different color ,  KHR_materials_unlit   and so on .  
![微信图片_20230608092951(1)](https://github.com/CesiumGS/cesium/assets/4468602/fb659867-141c-42bf-91dc-b493b7abaeb2)

as the new 3dtiles specification comes, 3dtiles next or 3dtiles 1.1 ,whatever we call it ,  we modified our tool to process point cloud data to glb with extension,  but when we load the glbs with the latest cesium, we found some problem in rendering and style setting,
after studying the CesiumJs code(1.105), we are confused  with what we found,
 1, model.js line from 3003 to 3010
constloader = new GltfLoader(loaderOptions):
is3DTiles = defined(options .content);constconsttype = is3DTiles ? ModelType.TILE GLTF: ModelType.GLTF;
const resource = loaderOptions.gltfResource;
const modelOptions = makeModelOptions(loader, type, options);

    the code above : we load glb with pointcloud data inside ,  there are two types: ModelType.TILE GLTF: ModelType.GLTF;

2 model.js  line: 3091 
  constmodel0ptions = makeModel0ptions(loader, ModelType.TILE PNTS, options);

  the code above: load pnts format,   we can the special typ for point cloud, ModelType.TILE PNTS. 

3  model.js  line 3170  
const isPnts = this,type === ModelType.TILE PNTS;

    we believe the problem is : the new glb format with point clould data, this new format,  no where to set ModelType.TILE PNTS.  and when CesiumJS renders point cloud data, it still needs this ModelType.TILE PNTS,

**Comment by geo3data** [comment]
[3dtiles_pnts_glb.zip](https://github.com/CesiumGS/cesium/files/11683254/3dtiles_pnts_glb.zip)
the data above is the new format glb with point could we processed, may give you some help.

**Comment by geo3data** [comment]
https://sandcastle.cesium.com/#c=jVJtb9MwEP4rp3yYEgk5DetgE13F1ExVSl8ozQadIiEvcVsvjh1sp1uC9t9xXgodAsSHKLbveZ67e+5cF8YScw0jomiRTVaA45goBVpAKQoJVHDAShGtIt5iUCA4SsgGF0xfNeBQpITDJUQWKSe7+3FMF3QS3FSBN6eBCvins3gUvAnS/MvtaHKBDOhbMk5rULp+vezPx5N0sfKyWXVHp+GynFeBXnxenq7LXjXzb7y5+e78dX/ux3Q6muR3Rmz2sCwX/kwFGdsl9T1cV/OHZbXwr/sz3kPb4jok3iZdBl/P8cX6nLLRZO3fTMOeersZP330ev4u3ZIP+7IfWe8iHguuNOwpeSTSdMLJY+cIum3e7MiKm/tIcI0pJzKyHMOLuI1VyWOwHbgcwveIA2hZtgeAVlZTRoyDRhc/YnrwGrW/Uz9sw2gjRWa8vardDhLbO+/1z7zTOkst1daGVEw4QbmkGdV0TxTCSWJ3CQ7QNktHqITIQnEMaUGuC1d5zkrQOwLdOEHpkhGgGzB88kRVPfVffZAnLbEybXRiqFkM1D53uQ3Xbk9w6NOIG8MSu8U5cHLyTwAyK/dfoMMSruqinRbuHJyHn0W2Pb2Y6bHzDfuvql1Xz/XvGWKs4x3YREohneMZC0YQE9su0pAM5dmx67P1yho0RQzr9/c0y4XUUEhmI+RqkuUMa6Lc+yJOTbmxUjVp4B4og4TugSaXf9hAiJkZgYlsCsZWtCKRNRy4Bv+CxgROKN8u9kQyXBpIXcZg5w2nbQAhNHDNtU76O1cLwe6xPNL9AQ
this is the sandcastle example,  
this is the first time  I sub a Issue, so  maybe messes , I will do better next time.

**Comment by geo3data** [comment]
![微信图片_20230608100838(1)](https://github.com/CesiumGS/cesium/assets/4468602/81a04c40-c702-4542-9d1c-091d9f25f304)
![微信图片_20230608101023(1)](https://github.com/CesiumGS/cesium/assets/4468602/bdaace96-3fda-45d0-9bac-8557f32fb29b)

the first picture is pnts,  is good. 
the second is glb (3dtiles next), the effect is bad,   we have set KHR_materials_unlit,  but all white .

**Comment by ggetz** [comment]
Thanks for the report @geo3data.

The Sandcastle example is very helpful. Thanks for including it in your report!

For some reason, I'm not seeing the point cloud show up at all.

As a first step, I would suggest you confirm your 3D Tiles data is valid. Please run it through the [3d-tiles-validator](https://github.com/CesiumGS/3d-tiles-validator) and let us know the results.

**Comment by lilleyse** [comment]
@geo3data `pnts` colors are by convention sRGB while glTF vertex colors are linear. In your tool you'll need to convert colors from sRGB to linear.

This is noted in the migration guide: https://github.com/CesiumGS/3d-tiles/blob/main/specification/TileFormats/glTF/MIGRATION.adoc#point-cloud-pnts

**Comment by geo3data** [comment]
> @geo3data `pnts` colors are by convention sRGB while glTF vertex colors are linear. In your tool you'll need to convert colors from sRGB to linear.
> 
> This is noted in the migration guide: https://github.com/CesiumGS/3d-tiles/blob/main/specification/TileFormats/glTF/MIGRATION.adoc#point-cloud-pnts

thanks a lot, @lilleyse @ggetz , "sRGB to linear" helps and works. 

but about style , it still confuses us,    look at the code below,   for pnts,  it is good,  but for gltf with point cloud using extension,  there is no way to set ModelType.TILE PNTS,
so we believe there must have some problem with seting styles of point cloud data stored in gltf , 

![微信图片_20230609110159(1)](https://github.com/CesiumGS/cesium/assets/4468602/39cb35bd-51aa-4df9-bde3-46107c563c1b)

**Comment by lilleyse** [comment]
@geo3data you're right, that check that should also account for glTF point clouds. If you want to try fixing it here's one possible approach:

* Store a `_hasPointCloud` variable in `ModelSceneGraph`.
* When the scene graph is initially constructed we can set this variable to true if any `runtimeNodes` have `runtimePrimitives` where `primitiveType === PrimitiveType.POINTS`.
* Then when setting the style we can check `_sceneGraph.hasPointCloud` instead of `isPnts`.
